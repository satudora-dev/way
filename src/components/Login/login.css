.github:hover {
  border: none;
  padding: 0;
  background-color: #BDBDBD;
}
