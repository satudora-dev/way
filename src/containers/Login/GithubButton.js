import "./GithubButton.css";

import React from 'react';

const GithubButton = ( {onClick} ) =>{
  const style = {
    github:{
      width: "80px",
      height: "80px",
      borderRadius: "50%",
      padding: 0,
      transition: "all .3s",
    },
    button: {
      padding: 0,
      border: "none",
      cursor: "pointer",
      backgroundColor: "white",
      transition: "all .3s",
    }
  }
  return(
    <div>
        <button onClick={() => {
          onClick();
        }} className="github" style={style.button}>
          <img className="github" style={style.github} src="./github.svg" alt={"img"}/>
          <h3>LOGIN</h3>
        </button>

    </div>
  );
}

export default GithubButton;
