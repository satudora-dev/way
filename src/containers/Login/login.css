.github:hover {
  border: none;
  border-radius: 50%;
  padding: 0;
  background-color: #BDBDBD;
}
