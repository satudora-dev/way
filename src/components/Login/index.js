import React, { Component } from 'react';
import "./login.css";
import SiteInfo from '../../components/SiteInfo';



function GithubButton(props){
  const style = {
    github:{
      width: "80px",
      height: "80px",
      "border-radius": "50%",
      padding: 0,
      transition: "all .3s",
    },
    button: {
      padding: 0,
      border: "none",
      cursor: "pointer",
      "background-color": "white",
      transition: "all .3s",
    }
  }
  return(
    <div>
        <button onClick={props.onClick} className="github" style={style.button}>
          <img className="github" style={style.github} src="./github.svg" />
          <h3>LOGIN</h3>
        </button>

    </div>
  );
}

class Login extends Component {

  render() {
    const style = {
      container: {
        "padding-top": "80px",
        "padding-bottom": "100px",
        width: "100%",
        "text-align": "center",
        position: "relative",
      }
    }
    return (
      <div className="Login">
        <div style={style.container }>
          <SiteInfo/>
        </div>
        <GithubButton
          onClick={() => {
            this.props.loginWithGithub();
          }}/>
      </div>
    );
  }
}

export default Login;
