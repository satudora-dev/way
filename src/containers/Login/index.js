import React, { Component } from 'react';
import {firebaseDB,firebaseAuth} from '../../firebase';
import IconButton from '@material-ui/core/IconButton';
import SvgIcon from '@material-ui/core/SvgIcon';
import {withRouter} from 'react-router-dom';
import "./login.css";

const ref = firebaseDB.ref('accounts');

function GithubButton(props){
  const style = {
    github:{
      width: "80px",
      height: "80px",
      "border-radius": "50%",
      padding: 0,
      transition: "all .3s",
    },
    button: {
      padding: 0,
      border: "none",
      cursor: "pointer",
      "background-color": "white",
      transition: "all .3s",
    }
  }
  return(
    <div>
        <button onClick={props.onClick} className="github" style={style.button}>
          <img className="github" style={style.github} src="./github.svg" />
          <h3>LOGIN</h3>
        </button>

    </div>
  );
}

class Login extends Component {
  constructor(props){
    super(props);
  }

  redirect(path,refKey){
    this.props.history.push({
      pathname: path,
    });
  }

  handleLogin(){//ポップアップログイン後にsignupに遷移
    const provider=new firebaseAuth.GithubAuthProvider();
    firebaseAuth().signInWithPopup(provider).then(result=>{
      if(result.credential!=null){
        var idRef;
        ref.orderByChild('email').equalTo(result.user.email)//メールアドレスが既に登録されているか
          .once('value',(snapshot)=>{
            if(snapshot.val()==null){//初回ログイン時
              idRef=ref.push();
              idRef.set({
                email: result.user.email,
                token: result.credential.accessToken,
                registered: false,
              }).then((result)=>{
                this.redirect('/signup',idRef);
              });
            }
            else{
              this.redirect('/signup',idRef);
            }
        });
      }
    });
  }

  render() {
    const style = {
      imgstyle: {
        "vertical-align": "middle",
        width: 256,
        height: 128,
      },
      container: {
        width: "100%",
        "line-height": "300px",
        "text-align": "center",
        position: "relative",
      }
    }
    return (
      <div className="Login">
        <div style={style.container}>
          <img src="./way.png" style={style.imgstyle}
            alt="loading the way image..." />
        </div>
        <GithubButton onClick={()=>this.handleLogin()}/>
      </div>
    );
  }
}

export default withRouter(Login);
