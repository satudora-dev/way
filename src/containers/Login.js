import Login from '../components/Login';
import { connect } from 'react-redux';

import { loginWithGithub }from '../actions/login';

const mapDispatchToProps = {
  loginWithGithub: loginWithGithub,
}

export default connect(null,mapDispatchToProps)(Login);
