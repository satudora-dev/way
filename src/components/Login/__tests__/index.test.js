import React from 'react';
import Login from '../../Login';
import renderer from 'react-test-renderer';

test('Login snapshot', () => {
  const component = renderer.create(
    <Login />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
