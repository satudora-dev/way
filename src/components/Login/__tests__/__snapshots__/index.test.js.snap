// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot 1`] = `
<div
  className="Login"
>
  <div
    className="sc-bwzfXH jlZvJh"
  >
    <div
      style={
        Object {
          "padding-top": "30px",
          "text-align": "center",
        }
      }
    >
      <img
        alt="loading the way image..."
        src="./way.png"
        style={
          Object {
            "height": 128,
            "vertical-align": "middle",
            "width": 256,
          }
        }
      />
      <h2>
        AI TOKYO LAB & Co.
      </h2>
    </div>
  </div>
  <div>
    <div
      className="sc-bdVaJa kMdoF"
      onClick={[Function]}
    >
      <img
        src="./github.svg"
      />
    </div>
    <h3>
      Login with GitHub
    </h3>
  </div>
</div>
`;
